### ISSUE TITLE
QueryIdGenerator stuck in unrecoverable state due to non-transactional execution of a statement.

### ISSUE THREAD

**Issue by stevenpyzhang** [confluentinc/ksql#4661]
**Describe the bug**
This happens because when we execute a statement in `InteractiveStatementExecutor`, we get to building the physical plan, the first action is to get the `queryId` from the `QueryIdGenerator`, create the Kafka Streams app, then set the next queryId in `InteractiveStatementExecutor`. If the Kafka Streams app creation exceptions out, then the QueryIdGenerator won't have a next id set and will be in an unrecoverable state until the server is restarted.

**To Reproduce**
Create a query that will fail when creating the KafkaStreams app. (For example: set the monitoring interceptor configs for consumers and producers created by KSQL as part of executing a query, but the jar file can't be found).

**Expected behavior**
We should be able to retry and attempt execution of the statement successfully.

**Actual behaviour**
We fail to execute the statement due to the QueryIdGenerator not having any available id set.
The KSQL logs gets spammed with this error

```
[2020-02-27 08:56:45,754] ERROR Stack trace: io.confluent.ksql.util.KsqlStatementException: QueryIdGenerator has not been updated with new offset
Statement: ...
	at io.confluent.ksql.engine.EngineExecutor.plan(EngineExecutor.java:213)
	at io.confluent.ksql.engine.KsqlEngine.plan(KsqlEngine.java:180)
	at io.confluent.ksql.rest.server.computation.InteractiveStatementExecutor.startQuery(InteractiveStatementExecutor.java:324)
	at io.confluent.ksql.rest.server.computation.InteractiveStatementExecutor.executeStatement(InteractiveStatementExecutor.java:275)
	at io.confluent.ksql.rest.server.computation.InteractiveStatementExecutor.handleStatementWithTerminatedQueries(InteractiveStatementExecutor.java:208)
	at io.confluent.ksql.rest.server.computation.InteractiveStatementExecutor.handleRestore(InteractiveStatementExecutor.java:131)
	at io.confluent.ksql.rest.server.computation.CommandRunner.lambda$null$1(CommandRunner.java:173)
	at io.confluent.ksql.util.RetryUtil.retryWithBackoff(RetryUtil.java:84)
	at io.confluent.ksql.util.RetryUtil.retryWithBackoff(RetryUtil.java:57)
	at io.confluent.ksql.util.RetryUtil.retryWithBackoff(RetryUtil.java:40)
	at io.confluent.ksql.rest.server.computation.CommandRunner.lambda$processPriorCommands$2(CommandRunner.java:169)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at io.confluent.ksql.rest.server.computation.CommandRunner.processPriorCommands(CommandRunner.java:166)
	at io.confluent.ksql.rest.server.KsqlRestApplication.initialize(KsqlRestApplication.java:342)
	at io.confluent.ksql.rest.server.KsqlRestApplication.startKsql(KsqlRestApplication.java:275)
	at io.confluent.ksql.rest.server.KsqlRestApplication.startAsync(KsqlRestApplication.java:260)
	at io.confluent.ksql.rest.server.ExecutableServer.startAsync(ExecutableServer.java:47)
	at io.confluent.ksql.rest.server.KsqlServerMain.tryStartApp(KsqlServerMain.java:75)
	at io.confluent.ksql.rest.server.KsqlServerMain.main(KsqlServerMain.java:61)
Caused by: io.confluent.ksql.util.KsqlServerException: QueryIdGenerator has not been updated with new offset
	at io.confluent.ksql.query.id.SpecificQueryIdGenerator.getNext(SpecificQueryIdGenerator.java:44)
	at io.confluent.ksql.planner.plan.KsqlStructuredDataOutputNode.getQueryId(KsqlStructuredDataOutputNode.java:96)
	at io.confluent.ksql.physical.PhysicalPlanBuilder.buildPhysicalPlan(PhysicalPlanBuilder.java:62)
	at io.confluent.ksql.engine.QueryEngine.buildPhysicalPlan(QueryEngine.java:105)
	at io.confluent.ksql.engine.EngineExecutor.planQuery(EngineExecutor.java:232)
	at io.confluent.ksql.engine.EngineExecutor.plan(EngineExecutor.java:182)
	... 18 more
 (io.confluent.ksql.util.RetryUtil:105)
```

**Additional context**
This isn't a problem in 0.8.0 and onwards because the queryId will be stored in the plans that are in the command topic.

**Comment by apurvam** [comment]
Did this make it into 0.7.1 @stevenpyzhang ?

**Comment by stevenpyzhang** [comment]
@apurvam unfortunately this didn't make it in. How high of a priority should this fix be?

**Comment by apurvam** [comment]
I don't think it is critical for 0.7.1. Is this fixed in 5.5?

**Comment by stevenpyzhang** [comment]
@apurvam I hadn't realized that we also turned on execution plans for 5.5, so this isn't actually a problem in 5.5. https://github.com/confluentinc/ksql/pull/4550

This bug only shows up when we're executing statements in the CommandRunner, which is what 0.7.0 does.
